packages/maskbook/src/plugins/RedPacket/UI/RedPacket.tsx

```diff
@@ -130,6 +144,9 @@ export function RedPacket(props: RedPacketProps) {
     const account = useAccount()
     const chainId = useChainId()
     const chainIdValid = useChainIdValid()
+    const wallet = useWallet(payload.sender.address)
+    const isMetamaskRedpacketLocked =
+        useValueRef(currentIsMetamaskLockedSettings) && wallet?.provider === ProviderType.MetaMask
```

Comment:
```suggestion

        useValueRef(currentIsMetamaskLockedSettings) && useValueRef(currentSelectedWalletProviderSettings)=== ProviderType.MetaMask
```

Reply:
No, you can't do this, this violates the React hooks rule.

Reply:
Notice